### DISCUSSION TITLE
How to change snapshots folder location? is it possible?

### DISCUSSION THREAD

**Discussion by andreparenteCorpGlobo** [pointfreeco/swift-snapshot-testing#559] [Q&A]

**Comment by GerritPost** [comment]
I've used  #filePath.replace to change the directory 

All other testclasses inheritance from SnapshotTestCase

```
import SnapshotTesting
import XCTest

class SnapshotTestCase: XCTestCase {
    
    var snapshotDirectory: String {
       return #filePath.replace("Tests/SnapshotTestCase.swift", withString: "Snapshot")
    }

    func testcaseSut() throws {
        ....
        XCTAssertNil(verifySnapshot(matching: sut, as: .image(on: .iPhoneX), record: isRecording, snapshotDirectory: snapshotDirectory))
    }
}
```

Using swift 5, SnapshotTesting 1.8.2

**Comment by andreparenteCorpGlobo** [comment]
@GerritPost where do you use this variable? on assertSnapshot function?

**Comment by andreparenteCorpGlobo** [comment]
the assertSnapshot receives a StaticString parameters, so I am not able to use replacing functions because it returns String object. Which version of Swift and of this lib are you using? @GerritPost

**Comment by GerritPost** [comment]
Updated the original answer, it was missing some imports :) Hope this helps you
